### PR TITLE
Assorted documentation

### PR DESCRIPTION
**Pull request by derekdreery**
I'm just trying to learn how this lib works, and I'm looking at source to work it out. Hopefully if I doc as I go, others that come after me will benefit :)

### CODE REVIEW

File: markup5ever/serialize.rs [hunk 1]
```diff
@@ -6,35 +6,64 @@
...
+//! implement their own traits.
+//!
+//! [processing instructions]: https://en.wikipedia.org/wiki/Processing_Instruction
 
 use QualName;
 use std::io;
 
 //§ serializing-html-fragments
+/// Used as a parameter to `serialize`, telling it if we want to skip the parent.
+/// TODO check: Currently the parameter to `ChildrenOnly` is unused.
```
Comment (jdm) [inline markup5ever/serialize.rs]: It's used by the HTML serializer code: https://github.com/derekdreery/html5ever/blob/30891d283147bfa815d1ae47694ff97761b2f150/html5ever/src/serialize/mod.rs#L77

File: markup5ever/serialize.rs [hunk 2]
```diff
@@ -6,35 +6,64 @@
...
 #[derive(Clone, PartialEq)]
 pub enum TraversalScope {
+    /// Include the parent node when serializing.
     IncludeNode,
+    /// Only serialize the children of the node.
```
Comment (jdm) [inline markup5ever/serialize.rs]: `/// Only serialize the children of the node, treating any provided qualified name as the parent while serializing.`

File: markup5ever/util/buffer_queue.rs [hunk 3]
```diff
@@ -123,6 +176,29 @@ impl BufferQueue {
     // If so, consume them and return Some(true).
     // If they do not match, return Some(false).
     // If not enough characters are available to know, return None.
+    /// Consume bytes matching the pattern, using a custom comparison function `eq`.
```
Comment (jdm) [inline markup5ever/util/buffer_queue.rs]: Let's remove the existing comments, since the new ones are more thorough.

**Comment by derekdreery** [comment]
I've addressed the issues you've raised. You're welcome to merge this, or if you leave it open I can continue adding documentation to it.

Or if you would rather I'll start a new PR with any other documentation I do. I don't mind either way.

**Comment by jdm** [comment]
Let's merge the changes that exist right now so others can benefit.
@bors-servo: r+

**Pull request merged**